### PR TITLE
fix(ordering): allow groups and associations in subprocesses

### LINKED ISSUE #1601: Error: no parent for <Group_0evfkzo> in <Activity_1wcp392_plane>
__Describe the Bug__

With https://github.com/bpmn-io/bpmn-io-demo/commit/4c004617b20893a1eeb85074b8e22fecaa9fe856 we seem to have introduced an error which can be seen via Sentry issue [BPMN-IO-DEMO-2AD](https://sentry.io/organizations/camunda-modeling/issues/3003618132/?referrer=github_integration)

__Steps to Reproduce__

unfortuently unclear yet

__Expected Behavior__

No new errors after version bump

__Environment__

See  [BPMN-IO-DEMO-2AD](https://sentry.io/organizations/camunda-modeling/issues/3003618132/?referrer=github_integration).

This was introduced with https://github.com/bpmn-io/bpmn-io-demo/commit/4c004617b20893a1eeb85074b8e22fecaa9fe856

__Context__

```
Error: no parent for <Group_0evfkzo> in <Activity_1wcp392_plane>
  at findActualParent (./node_modules/bpmn-js/lib/features/ordering/BpmnOrderingProvider.js:123:13)
  at getOrdering (./node_modules/bpmn-js/lib/features/ordering/BpmnOrderingProvider.js:142:19)
  at apply (./node_modules/bpmn-js/node_modules/diagram-js/lib/features/ordering/OrderingProvider.js:48:25)
  at invokeFunction (./node_modules/bpmn-js/node_modules/diagram-js/lib/core/EventBus.js:519:13)
  at _invokeListener (./node_modules/bpmn-js/node_modules/diagram-js/lib/core/EventBus.js:371:19)
...
(21 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by marstamm**
related to https://github.com/camunda/camunda-modeler/issues/2751, https://github.com/camunda/camunda-modeler/issues/2752

closes #1601, #1602

### CODE REVIEW

File: test/spec/features/ordering/BpmnOrderingProviderSpec.js [hunk 1]
```diff
@@ -343,6 +343,48 @@ describe('features/modeling - ordering', function() {
...
+
+      beforeEach(bootstrapModeler(diagramXML, { modules: testModules }));
+
+
+      describe('should stay always in front', function() {
+
+        it('moving <Group> onto <StartEvent>', inject(function() {
+
+          // when
+          move('Group', { x: 100, y: 0 }, 'StartEvent', false);
```
Comment (nikku) [inline test/spec/features/ordering/BpmnOrderingProviderSpec.js:359]: To consider: Add an integration test to ensure groups can be added in _collapsed_ sub-processes.

**Comment by nikku** [comment]
What is our expected behavior from the UX perspective?

What you're crafting here is: Move groups to the `flowElementsContainer? 

What I could imagine: Move groups always to the current root element.

**Comment by nikku** [comment]
Going with 

> What you're crafting here is: Move groups to the `flowElementsContainer?

Has undesired side-effects, such as groups moving the the parent at is being moved.

**Comment by marstamm** [comment]
@nikku, we already ensure groups can only be dropped on root elements: https://github.com/bpmn-io/bpmn-js/blob/6c02ecde34cf84b5686440391812bfdc7ec88adc/lib/features/modeling/behavior/FixHoverBehavior.js#L43-L48

So this PR just ensures that all roots are valid recipients.

File: test/spec/features/ordering/BpmnOrderingProviderSpec.js (hunk 1 above)
Comment (marstamm) [reply test/spec/features/ordering/BpmnOrderingProviderSpec.js:359]: Added with [39dd523](https://github.com/bpmn-io/bpmn-js/pull/1599/commits/39dd5235614fef390c73c9dc2f624d2ad3b402bd)

**Comment by MaxTru** [comment]
@nikku you were actively reviewing, so I assume you are on this. Let me know if not, and I will take a look.

**Comment by nikku** [comment]
I did integration test the behavior and can confirm that things do not break anymore.

However, I also was able to construct cases where inconsistent diagram state is being created.

As mentioned (https://github.com/bpmn-io/bpmn-js/pull/1599#issuecomment-1038985548) our existing behavior is that artifacts (groups, text annotations) are always moved to the current root. With this fix in place that behavior is missing, as I am expanding a sub-process in place.

The artifacts are still _inside_ the sub-process, both visually, but also as I am interacting with them. Moving one of them does not move the other (i.e. when groups are annotated).

This is just wired behavior I think. But as nothing breaks I'd leave this open as a future issue for our users to find / confirm.

**Review by nikku — APPROVED** [review]

**Pull request merged**